qemu/tests/migration_multi_host_with_kdump.py

```diff
@@ -0,0 +1,187 @@
+import logging
+from autotest.client.shared import error
+from autotest.client.shared.syncdata import SyncData
+from virttest import env_process
+from virttest import utils_test
+from generic.tests import kdump
+
+
+@error.context_aware
+def run(test, params, env):
+
+    """
+    KVM multi-host migration ping pong test:
+
+    Migration execution progress is described in documentation
+    for migrate method in class MultihostMigration.
+
+    The test procedure:
+    1) starts vm on master host.
+    2) Configure kdump inside guest
+    3) Implement kdump inside guest and pin the task to 1 cpu
+    4) do ping pong migration until guest was verified to reboot succeed
+    5) check "/var/crash" directory inside guest, whether dump file generates
+
+    :param test: kvm test object.
+    :param params: Dictionary with test parameters.
+    :param env: Dictionary with the test environment.
+    """
+
+    mig_protocol = params.get("mig_protocol", "tcp")
+    base_class = utils_test.qemu.MultihostMigration
+    if mig_protocol == "fd":
+        base_class = utils_test.qemu.MultihostMigrationFd
+    if mig_protocol == "exec":
+        base_class = utils_test.qemu.MultihostMigrationExec
+    if "rdma" in mig_protocol:
+        base_class = utils_test.qemu.MultihostMigrationRdma
```

Comment:
Unnecessary consideration for the mig_protocol, these three 'if' could be dropped.